Sources/Moya/Plugin.swift

```diff
@@ -13,18 +13,25 @@ public protocol PluginType {
     /// Called immediately before a request is sent over the network (or stubbed).
     func willSend(_ request: RequestType, target: TargetType)
 
+    /// Called when the request is ready to be sent. Return a StubBehavior if you want your request to be stubbed,
+    /// or `nil` to actually send the request over the network
+    ///
+    /// If a provider has more than 1 plugin returning a non nil value,only the value of the first of those plugins will be used.
+    func stubBehavior(for target: TargetType) -> StubBehavior?
```

Comment:
I like to think of plugins to be the centralized place to customize providers so that we don't have to search if we have to provide a closure, a plugin, or anything else. I agree the new stubBehavior(for:) function in PluginType doesn't perfectly fit with existing ones, mainly because values could be ignored.